starter-code/js/index.js

```diff
@@ -1,52 +1,102 @@
-function deleteItem(e){
-
-}
-
-function getPriceByProduct(itemNode){
-
-}
-
-function updatePriceByProduct(productPrice, index){
-
-}
-
-function getTotalPrice() {
-
-}
-
-function createQuantityInput(){
-
-}
-
-function createDeleteButton(){
-
+// Settings for the functions
+var masterContainer = document.getElementById("master-container-div");
+var containers = document.querySelectorAll(".div-container");
+var productName = document.querySelectorAll(".product-name");
+var productPrice = Number(
+  document.querySelectorAll(".product-price").textContent
+);
+var quantity = document.querySelectorAll(".quantity");
+var totalPrice = document.querySelectorAll(".total-price");
+var newProductName = document.querySelector(".new-product").textContent;
+var newProductPrice = Number(document.querySelector(".new-product-price"));
+var createNewItem = document.querySelector("#new-item-create");
+
+function wrapper() {
+  // var masterContainer = document.getElementById("master-container-div");
+  deleteItem(masterContainer);
+  getPriceByProduct(masterContainer);
+  getTotalPrice(masterContainer);
+  createItem(masterContainer);
 }
 
-function createQuantityNode(){
-
+createNewItem.addEventListener("click", wrapper);
+
+// delete Item from the button
+// newItem
+function createItem(masterContainer) {
+  var newItem = document.createElement("div");
+  newItem.className = "div-container";
+  masterContainer.append(newItem);
+  var newProductName = document.querySelector(".new-product").value;
+  console.log(newProductName);
+  var newProductPrice = document.querySelector(".new-product-price").value;
+  newItem.innerHTML += `<div>
+      <span class="product-name">${newProductName}</span>
+    </div>
+    <div>$<span class="product-price">${newProductPrice}</span></div>
+    <div>
+      <label for="quantity" class="qty">QTY</label>
+      <input
+        type="number"
+        name="quantity"
+        min="0"
+        value="0"
+        class="quantity"
+      />
+    </div>
+    <div><span class="total-price">$0</span></div>
+    <div><button class="btn btn-delete">Delete</button></div>`;
+  getPriceByProduct();
+  getTotalPrice();
+  deleteItem();
 }
-
-function createItemNode(dataType, itemData){
-
+// delete Item from the button
+function deleteItem() {
+  containers.forEach(function(container) {
+    var deleteButtons = container.querySelector(".btn-delete");
+    deleteButtons.addEventListener("click", function() {
+      container.remove(container);
+    });
+  });
 }
-
-function createNewItemRow(itemName, itemUnitPrice){
-
+// call the function
+deleteItem();
+
+// print the price for each item
+function getPriceByProduct() {
+  containers.forEach(function(container) {
+    var productPrice = container.querySelector(".product-price").textContent;
```

Comment:
The reason why the listeners on the delete button / updating the price of each elements aren't working is because you are referencing to the same array when the pages loads,
Here, the "containers" you are referencing are the first containers that are present on the page, so the node list only has 2 elements at that point.
What you need to do is just query selector once again the elements once you calculate the price, so you only to add this at the start of your function.
` var containers = document.querySelectorAll(".div-container");`